specs/ClearBrowsingData.md

```diff
@@ -0,0 +1,268 @@
+
+  /// Specifies DOM storage data. This browsing data kind is inclusive 
+  /// of COREWEBVIEW2_BROWSING_DATA_KINDS_APP_CACHE, 
+  /// COREWEBVIEW2_BROWSING_DATA_KINDS_FILE_SYSTEMS,
+  /// COREWEBVIEW2_BROWSING_DATA_KINDS_INDEXED_DB, 
+  /// COREWEBVIEW2_BROWSING_DATA_KINDS_LOCAL_STORAGE,
+  /// COREWEBVIEW2_BROWSING_DATA_KINDS_WEB_SQL, 
+  /// COREWEBVIEW2_BROWSING_DATA_KINDS_CACHE_STORAGE.
+  COREWEBVIEW2_BROWSING_DATA_KINDS_DOM_STORAGE = 
+      COREWEBVIEW2_BROWSING_DATA_KINDS_APP_CACHE|
+      COREWEBVIEW2_BROWSING_DATA_KINDS_FILE_SYSTEMS | 
+      COREWEBVIEW2_BROWSING_DATA_KINDS_INDEXED_DB |
+      COREWEBVIEW2_BROWSING_DATA_KINDS_LOCAL_STORAGE | 
+      COREWEBVIEW2_BROWSING_DATA_KINDS_WEB_SQL |
+      COREWEBVIEW2_BROWSING_DATA_KINDS_CACHE_STORAGE,
+
+  /// Specifies HTTP cookies data.
+  COREWEBVIEW2_BROWSING_DATA_KINDS_COOKIES = 1 << 6,
+
+  /// Specifies site data. This browsing data kind
+  /// is inclusive of COREWEBVIEW2_BROWSING_DATA_KINDS_DOM_STORAGE and
+  /// COREWEBVIEW2_BROWSING_DATA_KINDS_COOKIES. 
+  COREWEBVIEW2_BROWSING_DATA_KINDS_SITE = COREWEBVIEW2_BROWSING_DATA_KINDS_DOM_STORAGE | 
+      COREWEBVIEW2_BROWSING_DATA_KINDS_COOKIES, 
+
+  /// Specifies content in the HTTP cache including images and other files. 
+  COREWEBVIEW2_BROWSING_DATA_KINDS_HTTP_CACHE = 1 << 7,
+
+  /// Specifies download history data. 
+  COREWEBVIEW2_BROWSING_DATA_KINDS_DOWNLOAD_HISTORY = 1 << 8, 
+
+  /// Specifies general autofill form data. 
+  /// This excludes password information and includes information like: 
+  /// names, street and email addresses, phone numbers, and arbitrary input. 
+  /// This also includes payment data. 
+  COREWEBVIEW2_BROWSING_DATA_KINDS_GENERAL_AUTOFILL = 1 << 9, 
+
+  /// Specifies password autosave data. 
+  COREWEBVIEW2_BROWSING_DATA_KINDS_PASSWORD_AUTOSAVE = 1 << 10,
+
+  /// Specifies browsing history data. 
+  COREWEBVIEW2_BROWSING_DATA_KINDS_BROWSING_HISTORY = 1 << 11,
+
+  /// Specifies settings data.
+  COREWEBVIEW2_BROWSING_DATA_KINDS_SETTINGS = 1 << 12,
+
+  /// Specifies profile data that should be wiped to make it look like a new profile.
+  /// This browsing data kind is inclusive of COREWEBVIEW2_BROWSING_DATA_KINDS_SITE,
+  /// COREWEBVIEW2_BROWSING_DATA_KINDS_HTTP_CACHE, 
+  /// COREWEBVIEW2_BROWSING_DATA_KINDS_DOWNLOAD_HISTORY,
+  /// COREWEBVIEW2_BROWSING_DATA_KINDS_GENERAL_AUTOFILL, 
+  /// COREWEBVIEW2_BROWSING_DATA_KINDS_PASSWORD_AUTOSAVE,
+  /// COREWEBVIEW2_BROWSING_DATA_KINDS_BROWSING_HISTORY, and 
+  /// COREWEBVIEW2_BROWSING_DATA_KINDS_SETTINGS.
+  COREWEBVIEW2_BROWSING_DATA_KINDS_PROFILE =  
+    COREWEBVIEW2_BROWSING_DATA_KINDS_SITE |
+    COREWEBVIEW2_BROWSING_DATA_KINDS_HTTP_CACHE | 
+    COREWEBVIEW2_BROWSING_DATA_KINDS_DOWNLOAD_HISTORY |
+    COREWEBVIEW2_BROWSING_DATA_KINDS_GENERAL_AUTOFILL | 
+    COREWEBVIEW2_BROWSING_DATA_KINDS_PASSWORD_AUTOSAVE |
+    COREWEBVIEW2_BROWSING_DATA_KINDS_BROWSING_HISTORY | 
+    COREWEBVIEW2_BROWSING_DATA_KINDS_SETTINGS,
```

Comment:
Should we have an "ALL" value that covers any future browsing data that gets added in the future? (Cached aromas...)

That gives an app a way to simulate "delete the user data folder and then create a fresh one."

Reply:
Right. But what if the next version of WebView2 adds a new category of browsing data, with a new flag. `COREWEBVIEW2_BROWSING_DATA_KINDS_AROMAS`. Old code that wants to clear all data won't know to pass this new flag.

Should code that wants to "clear all browsing data, as if I had deleted the user data folder" just call ClearBrowsingData 32 times, once for each bit? Right now, they would have to rev each time a WebView2 is released with support for a new type of browsing data. Otherwise, they leak aroma data.

Reply:
Let's add an _ALL with clear documentation that it will clear all data even new kinds of data we add in the future.

Let's also add a ClearBrowsingData() method overload that takes no parameters and just clears all data.

Reply:
Is it OK to add an `all` enum value? The guidelines seem to suggest against it for compat reasons
https://www.osgwiki.com/wiki/WinRT_API_Design_Guidelines#Designing_Flag_Enums

> ❌ DO NOT name a flags enum value “All” unless the value actually contains all bits set.

Reply:
Don't need to _ALL if we just add the overload with no parameters. Let's do that!